### PR TITLE
port css and entry_completion from gtk-rs/examples4

### PR DESCRIPTION
**Pull request by vamsikrishna-brahmajosyula**

### CODE REVIEW

File: examples/src/bin/css.rs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+//! This sample demonstrates how to use CSS with gtk-rs.
+
+use gio::prelude::*;
+use gtk::prelude::*;
+
+use std::env::args;
+
+use gdk::Display;
+use gtk::{
+    Application, ApplicationWindow, Box, Button, ComboBoxText, CssProvider, Entry, Orientation,
```
Comment (bilelmoussaoui) [inline examples/src/bin/css.rs:12]: Rename Box to Box_ as `Box` is a pre-defined type.

File: examples/src/bin/css.rs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    let vbox = Box::new(Orientation::Vertical, 0);
+
+    let label = Button::with_label("hover me!");
+    // We need to name it in order to be able to use its name as a CSS label to
+    // apply CSS on it.
+    WidgetExtManual::set_name(&label, "label1");
+
+    let entry = Entry::new();
+    // We need to name it in order to apply CSS on it.
+    WidgetExtManual::set_name(&entry, "entry1");
```
Comment (bilelmoussaoui) [inline examples/src/bin/css.rs:59]: Not necessarly, use entry.add_class(".entry1") instead and you can match it with css using `entry.entry1{ color: red;}`

File: examples/src/bin/entry_completion.rs [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    (@param _) => ( _ );
+    (@param $x:ident) => ( $x );
+    ($($n:ident),+ => move |$($p:tt),*| $body:expr) => (
+        {
+            $( let $n = $n.clone(); )+
+                move |$(clone!(@param $p),)*| $body
+        }
+    );
+}
+
```
Comment (bilelmoussaoui) [inline examples/src/bin/entry_completion.rs:27]:
```suggestion
```
the macro exists in glib nowadays, so just use glib::clone!

File: examples/src/bin/entry_completion.rs [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        Default::default(),
+    )
+    .expect("Initialization failed...");
+    application.connect_activate(|app| {
+        build_ui(app);
+    });
+
+    // When activated, shuts down the application
+    let quit = SimpleAction::new("quit", None);
+    quit.connect_activate(clone!(application => move |_action, _parameter| {
```
Comment (bilelmoussaoui) [inline examples/src/bin/entry_completion.rs:110]:
```suggestion
    quit.connect_activate(glib::clone!(@weak application => move |_action, _parameter| {
```

**Comment by bilelmoussaoui** [comment]
Overall looks good to me, thanks!

File: examples/src/bin/css.rs (hunk 2 above)
Comment (vamsikrishna-brahmajosyula) [reply examples/src/bin/css.rs:59]: entry.add_class method does not exist. I see the same available in a StyleContext trait but button and entry both do not satisfy the same.
    let label = Button::with_label("hover me!");
    label.add_class("label1");

    let entry = Entry::new();
    entry.add_class(".entry1");
    entry.set_text("Some text");

Comment (bilelmoussaoui) [reply examples/src/bin/css.rs:59]: Oups, it's `add_css_class`

File: examples/src/bin/entry_completion.rs [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    completion_countries.set_minimum_key_length(1);
+    // whether the completions should be presented in a popup window
+    completion_countries.set_popup_completion(true);
+
+    // Create a ListStore of items
+    // These will be the source for the autocompletion
+    // as the user types into the field
+    // For a more evolved example of ListStore see src/bin/list_store.rs
+    let ls = create_list_model();
+    completion_countries.set_model(Some(&ls));
```
Comment (bilelmoussaoui) [inline examples/src/bin/entry_completion.rs:71]: You could use a gtk::StringList here which is a Model itself. 
```suggestion
    let countries = gtk::StringList::new(&["France", "Switzerland", "Italy"]);
    completion_countries.set_model(Some(&countries));
```
the create_list_model & `Data` won't be needed. Other might changes might be needed as well

File: examples/src/bin/css.rs [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    transition: 500ms;
+    color: red;
+    background-color: yellow;
+}
+
+combobox button.combo box {
+    padding: 5px;
+}
+combobox box arrow {
+    -gtk-icon-source: -gtk-icontheme('pan-down-symbolic');
```
Comment (bilelmoussaoui) [inline examples/src/bin/css.rs:38]: Is this still possible with gtk4?

File: examples/src/bin/entry_completion.rs (hunk 5 above)
Comment (vamsikrishna-brahmajosyula) [reply examples/src/bin/entry_completion.rs:71]: looks like the above is not enough..

a StringList is a Model but EntryCompletion expects a TreeModel.
"pub fn set_model<P: IsA<TreeModel>>"

How do I generate a TreeModel from StringList?

File: examples/src/bin/css.rs (hunk 6 above)
Comment (vamsikrishna-brahmajosyula) [reply examples/src/bin/css.rs:38]: I tried -gtk-icon-source: -gtk-icontheme('pan-up-symbolic'); (UP) and it seemed to work ok.

File: examples/src/bin/entry_completion.rs (hunk 5 above)
Comment (bilelmoussaoui) [reply examples/src/bin/entry_completion.rs:72]: Agh I thought ComboBox used the new fancy stuff....

**Pull request merged**